gcp_variant_transforms/bq_to_vcf.py

```diff
@@ -56,19 +63,42 @@ def run(argv=None):
   logging.info('Command: %s', ' '.join(argv or sys.argv))
   known_args, pipeline_args = vcf_to_bq_common.parse_args(argv,
                                                           _COMMAND_LINE_OPTIONS)
+  (project, dataset, table) = bigquery_util.parse_table_reference(
+      known_args.input_table)
   bq_source = bigquery.BigQuerySource(
       query=_BASE_QUERY_TEMPLATE.format(
-          INPUT_TABLE='.'.join(bigquery_util.parse_table_reference(
-              known_args.input_table))),
+          INPUT_TABLE='.'.join([project, dataset, table])),
       validate=True,
       use_standard_sql=True)
 
   options = pipeline_options.PipelineOptions(pipeline_args)
+
+  blob_prefix = '_sharded_vcf_files_{}'.format(
+      datetime.now().strftime('%Y%m%d_%H%M%S'))
+  file_path_prefix = '/'.join(['gs:/', known_args.bucket_name, blob_prefix])
   with beam.Pipeline(options=options) as p:
     _ = (p | 'ReadFromBigQuery ' >> beam.io.Read(bq_source)
          | bigquery_to_variant.BigQueryToVariant()
          | densify_variants.DensifyVariants()
-         | vcfio.WriteToVcf(known_args.output_file))
+         | 'PairVariantWithKey' >> beam.Map(_pair_variant_with_key,
+                                            _NUMBER_OF_VARIANTS_PER_SHARD)
+         | 'GroupVariantsByKey' >> beam.GroupByKey()
+         | vcfio.WriteVcfDataLines(file_path_prefix))
+
+  client = storage.Client(project)
+  bucket = client.get_bucket(known_args.bucket_name)
+  output_file_blob = bucket.blob(known_args.output_file)
+  output_file_blob.content_type = 'text/plain'
+  # Compose all VCF files generated in this pipeline into one.
+  output_file_blob.compose(bucket.list_blobs(prefix=blob_prefix))
```

Comment:
Note: There is a limit (currently 32) to the number of components that can be composed in a single operation for [compose](https://cloud.google.com/storage/docs/gsutil/commands/compose). To merge a large number of files, possible solutions can be found in [here](https://stackoverflow.com/questions/43024522/how-to-concatenate-a-million-files-in-google-cloud-storage), which I plan to try in my next PR.

Reply:
Updated the code.